result/result.py

```diff
@@ -1,211 +1,245 @@
-        is `Ok`.
+        Return the inner value
         """
-        if self._is_ok:
-            raise UnwrapError(message)
-        return cast(E, self._value)
+        return self._value
 
-    def unwrap(self) -> T:
+    def unwrap(self) -> NoReturn:
         """
-        Return the value if it is an `Ok` type. Raises an `UnwrapError` if it
-        is an `Err`.
+        Raises an `UnwrapError`.
         """
-        return self.expect("Called `Result.unwrap()` on an `Err` value")
+        raise UnwrapError("Called `Result.unwrap()` on an `Err` value")
 
     def unwrap_err(self) -> E:
         """
-        Return the value if it is an `Err` type. Raises an `UnwrapError` if it
-        is `Ok`.
+        Return the inner value
         """
-        return self.expect_err("Called `Result.unwrap_err()` on an `Ok` value")
+        return self._value
 
     def unwrap_or(self, default: T) -> T:
         """
-        Return the value if it is an `Ok` type. Return `default` if it is an
-        `Err`.
+        Return `default`.
         """
-        if self._is_ok:
-            return cast(T, self._value)
-        else:
-            return default
+        return default
 
     def map(self, op: Callable[[T], U]) -> 'Result[U, E]':
         """
-        If contained result is `Ok`, return `Ok` with original value mapped to
-        a new value using the passed in function. Otherwise return `Err` with
-        same value.
+        Return `Err` with the same value
         """
-        if not self._is_ok:
-            return cast(Result[U, E], self)
-        return Ok(op(cast(T, self._value)))
+        return cast(Result[U, E], self)
 
     def map_or(self, default: U, op: Callable[[T], U]) -> U:
         """
-        If contained result is `Ok`, return the original value mapped to a new
-        value using the passed in function. Otherwise return the default value.
+        Return the default value
         """
-        if not self._is_ok:
-            return default
-        return op(cast(T, self._value))
+        return default
 
     def map_or_else(
-        self,
-        default_op: Callable[[], U],
-        op: Callable[[T], U]
+            self,
+            default_op: Callable[[], U],
+            op: Callable[[T], U]
     ) -> U:
         """
-        If contained result is `Ok`, return original value mapped to
-        a new value using the passed in `op` function. Otherwise use `default_op`
-        to compute a default value.
+        Return the result of the default operation
         """
-        if not self._is_ok:
-            return default_op()
-        return op(cast(T, self._value))
+        return default_op()
 
     def map_err(self, op: Callable[[E], F]) -> 'Result[T, F]':
         """
-        If contained result is `Err`, return `Err` with original value mapped
-        to a new value using the passed in `op` function. Otherwise return `Ok`
-        with the same value.
+        Return `Ok` with the same value
```

Comment:
Wrong docstring.

Reply:
Fixed in 5496f6bbf119ddb150776c72c2ac842e28f72b00